File: performance/implementation-Python/partition.py
from typing import Tuple


def partition(in_str: str, partition_str: str) -> Tuple[str, str, str]:
    max_idx_partition = len(in_str) - len(partition_str)

    i = 0
    while i <= max_idx_partition:
        substr = in_str[i:i + len(partition_str)]
        if substr == partition_str:
            break
        i += 1

    if i <= max_idx_partition:
        return in_str[:i], partition_str, in_str[i + len(partition_str):]

    return in_str, "", ""

File: performance/implementation-C/partition.cpp
#include "partition.hpp"
#include <iostream>


std::tuple<std::string, std::string, std::string> partition(const std::string& inStr, const std::string partitionStr){
    const long max_idx_partition = inStr.length() - partitionStr.length();

    long i = 0;
    while(i <= max_idx_partition){
        const std::string substr = inStr.substr(i, partitionStr.length());
        if (substr == partitionStr){
            break;
        }
        ++i;
    }

    if (i <= max_idx_partition){
        return std::make_tuple(inStr.substr(0, i), partitionStr, inStr.substr(i + partitionStr.length()));
    }

    return std::make_tuple(inStr, "", "");
}